### PR TITLE
refactor: clean up types in jest.setup.ts

### PR DESCRIPTION
**Pull request by jsjoeio**
This PR cleans up some of the types in `jest.setup.ts`.

### CODE REVIEW

File: site/jest.setup.ts [hunk 1]
```diff
@@ -38,16 +38,12 @@ afterAll(() => server.close())
 // For now, I limited this to just 'error' - but failing on warnings
 // would be a nice next step! We may need to filter out some noise
 // from material-ui though.
-const CONSOLE_FAIL_TYPES = ["error" /* 'warn' */]
+const CONSOLE_FAIL_TYPES = ["error" /* 'warn' */] as const
```
Comment (jsjoeio) [inline site/jest.setup.ts:41]: Using `as const` narrows the type to string literals which is more correct.

File: site/jest.setup.ts [hunk 2]
```diff
@@ -38,16 +38,12 @@ afterAll(() => server.close())
...
 
 // Throw errors when a `console.error` or `console.warn` happens
 // by overriding the functions
-CONSOLE_FAIL_TYPES.forEach((logType: string) => {
-  // Suppressing the no-explicit-any to override certain console functions for testing
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const consoleAsAny = global.console as any
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  consoleAsAny[logType] = (format: string, ...args: any[]): void => {
+CONSOLE_FAIL_TYPES.forEach((logType: typeof CONSOLE_FAIL_TYPES[number]) => {
```
Comment (jsjoeio) [inline site/jest.setup.ts:45]: here `logType` is limited to the options we specify in `CONSOLE_FAIL_TYPES`.

File: site/jest.setup.ts [hunk 3]
```diff
@@ -38,16 +38,12 @@ afterAll(() => server.close())
...
+  global.console[logType] = <Type>(format: string, ...args: Type[]): void => {
```
Comment (jsjoeio) [inline site/jest.setup.ts:46]: Using the generic, we no longer need to use `any`.

**Review by Kira-Pilot — APPROVED** [review]
Nice job! Thanks for the comments.

**Pull request merged**